src/asami/entities/writer.cljc

```diff
@@ -82,7 +82,7 @@
 (defn lookup-ref?
   "Tests if i is a lookup ref"
   [i]
-  (and (vector? i) (keyword? (first i)) (= 2 (count i))))
+  (and (vector? i) (= 2 (count i)) (#{:db/id :db/ident :id} (first i))))
```

Comment:
Lookup refs can actually be on any field in Datomic, and I want to keep that behavior
https://blog.datomic.com/2014/02/datomic-lookup-refs.html

Reply:
Thank you! I will fix that.

Reply:
FYI I have removed the set but replaced it with keyword?` - otherwise many non-refs pass as refs and a number of tests fails.